src/rounding.rs

```diff
@@ -0,0 +1,279 @@
+use crate::{
+    options::{TemporalRoundingMode, TemporalUnsignedRoundingMode},
+    TemporalError, TemporalResult,
+};
+
+use std::{
+    cmp::Ordering,
+    ops::{Div, Neg},
+};
+
+use num_traits::{ConstZero, Euclid, FromPrimitive, NumCast, Signed, ToPrimitive};
+
+pub(crate) trait Roundable:
+    Euclid + Div + PartialOrd + Signed + FromPrimitive + ToPrimitive + NumCast + ConstZero + Copy
+{
+    fn is_exact(dividend: &Self, divisor: &Self) -> bool;
+    fn compare_remainder(dividend: &Self, divisor: &Self) -> Option<Ordering>;
+    fn is_even_cardinal(dividend: &Self, divisor: &Self) -> bool;
+    fn result_floor(dividend: &Self, divisor: &Self) -> u64;
+    fn result_ceil(dividend: &Self, divisor: &Self) -> u64;
+    fn quotient_abs(dividend: &Self, divisor: &Self) -> Self {
```

Comment:
All of those could take Self by value instead, since they require `Copy`. That would simplify the API a bit.

Reply:
Initially, I had those as value because of the copy, but wouldn't copying `i128`s be more expensive then passing the reference?

Reply:
I'm pretty sure it's more costly to follow a pointer than to copy 8 additional bytes (additional indirection, the i128s won't be in the same cache line, etc.), but we can pick the simpler API and then benchmark in the future

Reply:
The change back isn't too difficult. I can change it to pass by value.